certbot/certbot/_internal/plugins/disco.py

```diff
@@ -44,15 +48,15 @@ class PluginEntryPoint:
     # this object is mutable, don't allow it to be hashed!
     __hash__ = None  # type: ignore
 
-    def __init__(self, entry_point, with_prefix=False):
+    def __init__(self, entry_point: pkg_resources.EntryPoint, with_prefix=False):
         self.name = self.entry_point_to_plugin_name(entry_point, with_prefix)
-        self.plugin_cls = entry_point.load()
+        self.plugin_cls: Any = entry_point.load()
```

Comment:
If I change this to `interfaces.IPluginFactory` which is what this object should be, it almost works, however, `mypy` complains about the number of parameters when we call `self.plugin_cls(...)`.

Do you think it's worth either of:

1. Using that interface as the type here and either fixing it or adding `type: ignore` lines to make `mypy` happy?
2. Adding a comment saying what this type can be in the future when we're off of `zope`.

Reply:
Option 1. Once we have removed the interface because we moved away from zope, this code will need a fix so we cannot miss the occasion to improve the type definition here.